### PR TITLE
Don't allow window zoom when CanResize=false and custom window chrome

### PR DESCRIPTION
**Pull request by grokys**
## What does the pull request do?

Following on from #10575, disable window zoom/resize when `CanResize == false` in two other scenarios when `ExtendClientAreaToDecorationsHint == true`:

- On macOS: double clicking the titlebar area to zoom was handled in `WindowImpl` but didn't check the `CanResize` state
- On Windows: The maximize button in `CaptionButtons` wasn't being disabled when `CanResize == false`

Moved the macOS-specific integration test for this case to run also on Windows, and extended the test to also test when `ExtendClientAreaToDecorationsHint `.

In addition fixed a 🤦 in `ControlAutomationPeer` that meant that setting `AutomationProperties.Name` wasn't working (which was needed for the integration test).

### CODE REVIEW

**Review by maxkatz6 — APPROVED** [review]

File: src/Avalonia.Themes.Fluent/Controls/CaptionButtons.xaml [hunk 1]
```diff
@@ -48,15 +48,17 @@
...
+                  AutomationProperties.Name="Minimise">
             <Viewbox Width="11" Margin="2">
               <Path Stretch="UniformToFill"
                     Fill="{TemplateBinding Foreground}"
                     Data="M2048 1229v-205h-2048v205h2048z" />
             </Viewbox>
           </Button>
           <Button x:Name="PART_RestoreButton"
-                  Theme="{StaticResource FluentCaptionButton}">
+                  Theme="{StaticResource FluentCaptionButton}"
```
[the remark this answers is not in the document]
Comment (grokys) [reply src/Avalonia.Themes.Fluent/Controls/CaptionButtons.xaml:60]: Hmm strange, I thought I also changed the default theme. Will put that back in.

[the remark this answers is not in the document]
Comment (grokys) [reply src/Avalonia.Themes.Fluent/Controls/CaptionButtons.xaml:60]: Done.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0032628-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by maxkatz6** [comment]
@grokys seems like windows integration tests are failing

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0032824-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0032832-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by grokys** [comment]
> seems like windows integration tests are failing

Strange, looks like it's unable to find the maximize button on the CI machine, though it can find it here.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0032837-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0032841-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0032953-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0032975-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0033066-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0033074-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**